From 8aaed833385d070cc3a4d05115bffd167cd4e6af Mon Sep 17 00:00:00 2001
From: Patric Stout <truebrain@openttd.org>
Date: Sat, 4 Dec 2021 18:32:06 +0100
Subject: [PATCH 0776/1053] Codechange: ensure OnConnect() always gets called
 with a valid socket (#9729)

This should already be the case, but now assert()s will tell us
if this isn't.
---
 src/network/core/tcp.cpp            | 2 ++
 src/network/core/tcp_connect.cpp    | 2 ++
 src/network/network_coordinator.cpp | 2 ++
 src/network/network_turn.cpp        | 2 +-
 4 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/src/network/core/tcp.cpp b/src/network/core/tcp.cpp
index a2e31b53d2..3dc2a3fac8 100644
--- a/src/network/core/tcp.cpp
+++ b/src/network/core/tcp.cpp
@@ -217,6 +217,8 @@ Packet *NetworkTCPSocketHandler::ReceivePacket()
  */
 bool NetworkTCPSocketHandler::CanSendReceive()
 {
+	assert(this->sock != INVALID_SOCKET);
+
 	fd_set read_fd, write_fd;
 	struct timeval tv;
 
diff --git a/src/network/core/tcp_connect.cpp b/src/network/core/tcp_connect.cpp
index 73c6aa90d3..f9c40c2026 100644
--- a/src/network/core/tcp_connect.cpp
+++ b/src/network/core/tcp_connect.cpp
@@ -451,6 +451,8 @@ bool TCPServerConnecter::CheckActivity()
  */
 void TCPServerConnecter::SetConnected(SOCKET sock)
 {
+	assert(sock != INVALID_SOCKET);
+
 	this->socket = sock;
 	this->status = Status::CONNECTED;
 }
diff --git a/src/network/network_coordinator.cpp b/src/network/network_coordinator.cpp
index c3fa2b5ce5..a52ac638ea 100644
--- a/src/network/network_coordinator.cpp
+++ b/src/network/network_coordinator.cpp
@@ -566,6 +566,8 @@ void ClientNetworkCoordinatorSocketHandler::ConnectFailure(const std::string &to
  */
 void ClientNetworkCoordinatorSocketHandler::ConnectSuccess(const std::string &token, SOCKET sock, NetworkAddress &address)
 {
+	assert(sock != INVALID_SOCKET);
+
 	/* Connecter will destroy itself. */
 	this->game_connecter = nullptr;
 
diff --git a/src/network/network_turn.cpp b/src/network/network_turn.cpp
index ae82f3094d..bfc4919e59 100644
--- a/src/network/network_turn.cpp
+++ b/src/network/network_turn.cpp
@@ -41,7 +41,7 @@ public:
 	{
 		this->handler->connecter = nullptr;
 
-		handler->sock = s;
+		this->handler->sock = s;
 	}
 };
 
-- 
2.55.0

